# Simulation settings for using moveit_sim_controllers
moveit_sim_hw_interface:
  joint_model_group: controllers_initial_group_
  joint_model_group_pose: controllers_initial_pose_
# Settings for ros_control_boilerplate control loop
generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01
# Settings for ros_control hardware interface
hardware_interface:
  joints:
    - l_shoulder_pitch
    - l_shoulder_roll
    - l_arm_yaw
    - l_elbow_pitch
    - l_forearm_yaw
    - l_wrist_pitch
    - l_wrist_roll
    - l_gripper
    - orbita_roll
    - orbita_pitch
    - orbita_yaw
    - l_antenna
    - r_antenna
    - r_shoulder_pitch
    - r_shoulder_roll
    - r_arm_yaw
    - r_elbow_pitch
    - r_forearm_yaw
    - r_wrist_pitch
    - r_wrist_roll
    - r_gripper
  sim_control_mode: 1  # 0: position, 1: velocity
# Publish all joint states
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50
controller_list:
  - name: right_arm_position_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - r_shoulder_pitch
      - r_shoulder_roll
      - r_arm_yaw
      - r_elbow_pitch
      - r_forearm_yaw
      - r_wrist_pitch
      - r_wrist_roll
  - name: left_arm_position_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - l_shoulder_pitch
      - l_shoulder_roll
      - l_arm_yaw
      - l_elbow_pitch
      - l_forearm_yaw
      - l_wrist_pitch
      - l_wrist_roll
  - name: head_controller
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - orbita_roll
      - orbita_pitch
      - orbita_yaw
right_arm_position_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - r_shoulder_pitch
    - r_shoulder_roll
    - r_arm_yaw
    - r_elbow_pitch
    - r_forearm_yaw
    - r_wrist_pitch
    - r_wrist_roll
  gains:
    r_shoulder_pitch:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    r_shoulder_roll:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    r_arm_yaw:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    r_elbow_pitch:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    r_forearm_yaw:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    r_wrist_pitch:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    r_wrist_roll:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
left_arm_position_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - l_shoulder_pitch
    - l_shoulder_roll
    - l_arm_yaw
    - l_elbow_pitch
    - l_forearm_yaw
    - l_wrist_pitch
    - l_wrist_roll
  gains:
    l_shoulder_pitch:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    l_shoulder_roll:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    l_arm_yaw:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    l_elbow_pitch:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    l_forearm_yaw:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    l_wrist_pitch:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    l_wrist_roll:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
head_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - orbita_roll
    - orbita_pitch
    - orbita_yaw
  gains:
    orbita_roll:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    orbita_pitch:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1
    orbita_yaw:
      p: 1000
      d: 0.1
      i: 1
      i_clamp: 1